openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java

```diff
@@ -0,0 +1,250 @@
+        SSLUtil sslUtil = new SSLUtil(trustManager);
+
+        LDAPConnectionOptions connectionOptions = new LDAPConnectionOptions();
+        SSLSocketVerifier sslSocketVerifier =
+            ldapConfiguration.getVerifyCertificateHostname()
+                ? new HostNameSSLSocketVerifier(true)
+                : TrustAllSSLSocketVerifier.getInstance();
+        connectionOptions.setSSLSocketVerifier(sslSocketVerifier);
+
+        try (LDAPConnection connection =
+            new LDAPConnection(
+                sslUtil.createSSLSocketFactory(),
+                connectionOptions,
+                ldapConfiguration.getHost(),
+                ldapConfiguration.getPort())) {
+          // Use the connection here.
+          return new LDAPConnectionPool(connection, ldapConfiguration.getMaxPoolSize());
+        } catch (GeneralSecurityException e) {
+          LOG.warn("[LDAP] Issue in creating a LookUp Connection SSL");
+        }
+      } else {
+        LDAPConnection conn =
+            new LDAPConnection(
+                ldapConfiguration.getHost(),
+                ldapConfiguration.getPort(),
+                ldapConfiguration.getDnAdminPrincipal(),
+                ldapConfiguration.getDnAdminPassword());
+        return new LDAPConnectionPool(conn, ldapConfiguration.getMaxPoolSize());
+      }
+    } catch (LDAPException e) {
+      LOG.warn("[LDAP] Issue in creating a LookUp Connection");
+    }
+    return null;
+  }
+
+  @Override
+  public JwtResponse loginUser(LoginRequest loginRequest) throws IOException, TemplateException {
+    // validate Login is Not Blocked
+    checkIfLoginBlocked(loginRequest.getEmail());
+    // Fetch the User from Database
+    User storedUser = lookUserInProvider(loginRequest.getEmail());
+    // validate User's Password
+    validatePassword(storedUser, loginRequest.getPassword());
+    // check db if OM db contains user else create
+    User omUser = checkAndCreateUser(loginRequest.getEmail());
+    // Return a Jwt Response
+    return getJwtResponse(omUser);
+  }
+
+  private User checkAndCreateUser(String email) throws IOException {
+    // Check if the user exists in OM Database
+    try {
+      return userRepository.getByName(null, email.split("@")[0], userRepository.getFields("id,name,email"));
+    } catch (EntityNotFoundException ex) {
+      // User does not exist
+      return userRepository.create(null, getUserForLdap(email));
+    }
+  }
+
+  @Override
+  public void checkIfLoginBlocked(String email) {
+    if (loginAttemptCache.isLoginBlocked(email)) {
+      throw new AuthenticationException(MAX_FAILED_LOGIN_ATTEMPT);
+    }
+  }
+
+  @Override
+  public void recordFailedLoginAttempt(User storedUser) throws TemplateException, IOException {
+    loginAttemptCache.recordFailedLogin(storedUser.getEmail());
+    int failedLoginAttempt = loginAttemptCache.getUserFailedLoginCount(storedUser.getEmail());
+    if (failedLoginAttempt == 3 /*TODO : take it from config*/) {
```

Comment:
can we fix the TODOs looks like we read it for basic auth but not for LDAP. This will be confusing for anyone during the support as we take some from config but not others